### PR TITLE
Create naive code generated version of ExecQual

### PR DESCRIPTION
**Pull request by armenatzoglou**
This PR generates code for a naive version of ExecQual that simply fallbacks to regular ExecQual. The code-generated ExecQual is enrolled in the existing codegen framework and is called only in ExecScan. After the implementation of the function (i.e., generate code for expression evaluation), we can replace all calls to regular ExecQual with call_ExecQual macros.

@hardikar @karthijrk @gcaragea @foyzur: Please take a look if you have time. Thanks

### CODE REVIEW

File: src/backend/codegen/CMakeLists.txt [hunk 1]
```diff
@@ -165,6 +165,7 @@ set(GPCODEGEN_SRC
     codegen_manager.cc
     codegen_wrapper.cc
     ExecVariableList_codegen.cc
+    ExecQual_codegen.cc
```
Comment (karthijrk) [inline src/backend/codegen/CMakeLists.txt]: Minor : File name should all be lower case. I just noticed that for ExecVariableList_codegen as well.

File: src/backend/codegen/ExecQual_codegen.cc [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+extern "C" {
+#include "postgres.h"
+#include "utils/elog.h"
+}
+
+using gpcodegen::ExecQualCodegen;
+
+constexpr char ExecQualCodegen::kExecQualPrefix[];
+
+class ElogWrapper {
```
Comment (karthijrk) [inline src/backend/codegen/ExecQual_codegen.cc]: Can you reuse this from ExecVarList_codegen itself.

File: src/backend/codegen/ExecQual_codegen.cc [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+
+bool ExecQualCodegen::GenerateExecQual(
+    gpcodegen::CodegenUtils* codegen_utils) {
+
+  assert(NULL != codegen_utils);
+
+  ElogWrapper elogwrapper(codegen_utils);
+
+  llvm::Function* ExecQual_func = codegen_utils->
```
Comment (karthijrk) [inline src/backend/codegen/ExecQual_codegen.cc]: variable names needs to be in lower case

File: src/backend/executor/execProcnode.c [hunk 4]
```diff
@@ -753,6 +753,15 @@ ExecInitNode(Plan *node, EState *estate, int eflags)
 
 	if (result != NULL)
 	{
+
+#ifdef USE_CODEGEN
+		// Set the default location for ExecQual
+		result->ExecQual_gen_info.ExecQual_fn = ExecQual;
```
Comment (karthijrk) [inline src/backend/executor/execProcnode.c]: Why not set always even if USE_CONDEGEN is not defined

**Comment by karthijrk** [comment]
Great work 👍  Except few minor change, they LGTM.

**Comment by gcaragea** [comment]
LGTM
(except for the duplication of the elog wrapper.. )

**Comment by armenatzoglou** [comment]
@karthijrk @gcaragea Thanks a lot for the review. Regarding the duplication of elog wrapper, I do not like it either, but this is not part of this story. We have prioritized a story at QX's backlog (#118417849) to fix this.

File: src/backend/executor/execProcnode.c (hunk 4 above)
Comment (armenatzoglou) [inline src/backend/executor/execProcnode.c]: Good catch! Thanks!

File: src/backend/codegen/CMakeLists.txt (hunk 1 above)
Comment (armenatzoglou) [inline src/backend/codegen/CMakeLists.txt]: Good catch. I will change it. ExecVariableList_codegen has been changed in dce266adac5569927d9bf566141aaea2e9cf16f7.

File: src/backend/codegen/ExecQual_codegen.cc (hunk 3 above)
Comment (armenatzoglou) [inline src/backend/codegen/ExecQual_codegen.cc]: I will do it

**Comment by armenatzoglou** [comment]
Thanks a lot for the review! Pushed to master aae0ad3d1ddfdd9c6284dc19bf996ed11da8e849.

**Pull request closed without merging**